torch/csrc/jit/register_special_ops.cpp

```diff
@@ -48,6 +48,20 @@ void checkListInputType(const c10::TypePtr& elem_type, const Node* node) {
   }
 }
 
+at::Tensor castTensorTo(at::Tensor self,
+                        const IValue& dtype,
+                        const IValue& device,
+                        at::ScalarType initial_scalar_type) {
```

Comment:
i think you can just use can just use the `self.scalar_type()` don't need `initial_scalar_type`...

Reply:
Yeah I think it should be the same, but seems like `torch.tensor` also use initial_scalar_type that is getting from `scalarTypeFromJitType`, did you remember why we have this?

Reply:
eh doesn't really matter. the tensor was constructed from `initial_scalar_type` so it and `self.scalar_type()` should be same value